pkg/core/resources/apis/meshexternalservice/hostname/generator_test.go

```diff
@@ -0,0 +1,193 @@
+		)
+		Expect(err).ToNot(HaveOccurred())
+		ch := make(chan struct{})
+		var stopChRecv <-chan struct{}
+		stopChSend, stopChRecv = ch, ch
+		go func() {
+			defer GinkgoRecover()
+			Expect(allocator.Start(stopChRecv)).To(Succeed())
+		}()
+
+		Expect(samples.MeshDefaultBuilder().Create(resManager)).To(Succeed())
+
+		generator := hostnamegenerator_api.NewHostnameGeneratorResource()
+		generator.Meta = &test_model.ResourceMeta{
+			Mesh: core_model.DefaultMesh,
+			Name: "byname",
+		}
+		generator.Spec = &hostnamegenerator_api.HostnameGenerator{
+			Template: "{{ .Name }}.byname.mesh",
+			Selector: hostnamegenerator_api.Selector{
+				MeshExternalService: hostnamegenerator_api.NameLabelsSelector{
+					MatchName: "test-external-svc",
+				},
+			},
+		}
+		Expect(resManager.Create(context.Background(), generator, store.CreateBy(core_model.MetaToResourceKey(generator.GetMeta())))).To(Succeed())
+		generator = hostnamegenerator_api.NewHostnameGeneratorResource()
+		generator.Meta = &test_model.ResourceMeta{
+			Mesh: core_model.DefaultMesh,
+			Name: "example",
+		}
+		generator.Spec = &hostnamegenerator_api.HostnameGenerator{
+			Template: "{{ .Name }}.mesh",
+			Selector: hostnamegenerator_api.Selector{
+				MeshExternalService: hostnamegenerator_api.NameLabelsSelector{
+					MatchLabels: map[string]string{
+						"label": "value",
+					},
+				},
+			},
+		}
+		Expect(resManager.Create(context.Background(), generator, store.CreateBy(core_model.MetaToResourceKey(generator.GetMeta())))).To(Succeed())
+		generator = hostnamegenerator_api.NewHostnameGeneratorResource()
+		generator.Meta = &test_model.ResourceMeta{
+			Mesh: core_model.DefaultMesh,
+			Name: "static",
+		}
+		generator.Spec = &hostnamegenerator_api.HostnameGenerator{
+			Template: "static.mesh",
+			Selector: hostnamegenerator_api.Selector{
+				MeshExternalService: hostnamegenerator_api.NameLabelsSelector{
+					MatchLabels: map[string]string{
+						"generate": "static",
+					},
+				},
+			},
+		}
+		Expect(resManager.Create(context.Background(), generator, store.CreateBy(core_model.MetaToResourceKey(generator.GetMeta())))).To(Succeed())
+	})
+
+	AfterEach(func() {
+		close(stopChSend)
+		Expect(resManager.DeleteAll(context.Background(), &meshexternalservice_api.MeshExternalServiceResourceList{})).To(Succeed())
+	})
+
+	vipOfMeshExternalService := func(name string) *meshexternalservice_api.MeshExternalServiceStatus {
+		ms := meshexternalservice_api.NewMeshExternalServiceResource()
+		err := resManager.Get(context.Background(), ms, store.GetByKey(name, model.DefaultMesh))
+		Expect(err).ToNot(HaveOccurred())
+		return ms.Status
+	}
+
+	It("should not generate hostname if no generator selects a given MeshExternalService", func() {
+		// when
+		err := samples.MeshExternalServiceExampleBuilder().WithoutVIP().Create(resManager)
+		Expect(err).ToNot(HaveOccurred())
+
+		// then
+		Eventually(func(g Gomega) {
```

Comment:
instead of Eventually shouldn't we use Consistently (+ must pass repeatedly)  in this case?